### PR TITLE
Search courses by name, number, and subject

### PR DESCRIPTION
**Pull request by VKong6019**
Issue: https://github.com/sandboxnu/pharmd-tracker/issues/44

Should fix the search functionality for searching courses and uses the "ILIKE" operator and "%" wildcard to search for course `name` and `subject`. The course `number` is queried by an exact match.

also uses the `name_like` query param for all these search queries

### CODE REVIEW

**Comment by sophiacrennan** [comment]
To get `_start` to return something, I have to give it a negative number. For example, `http://localhost:3000/courses?_start=-2 `returns two courses but  `http://localhost:3000/courses?_start=2` returns an error. Is `_start` only meant to work with `_end` (`http://localhost:3000/courses?_start=1&_end=2` returns one course)?

**Comment by VKong6019** [comment]
so `_start` and `_end` are the params set up from the frontend to handle pagination, which is setup in the endpoint here like this
```
.limit(end - start)
```
 `_start` should be the nth record where the endpoint will return the number of courses by the difference above after it.
personally, i think the params could be made easier to understand if `_end` was something more like `page` or `numberOfCourses` so that both `_start` and `_end` wouldn't be needed together. I also just realized that by default, having `_start` and `_end` set to 0 should return an empty list but it seems to return all the courses LOL.

**TLDR**: both `_start` and `_end` need to be used if someone wants to use either and the difference of those will return courses up to that difference

i'll make a note of this in the docs :P

**Review by sophiacrennan — APPROVED** [review]
